### PR TITLE
#134249129 login with jwt

### PR DESCRIPTION
**Pull request by andela-emurphy**
#### What does this PR do?             
This PR ensures that once a user successfully login,  they receive a JSON web token for further authentication.
#### Description of Task to be completed?   
When  a user successfully login, a json web token is sent back to that user. The token sent back is used to access restricted routes. 
#### How should this be manually tested?     
This PR can be tested manually by running a local server and using POSTMAN. User should send their valid details (username and password) to this endpoint "localhost:port/api/auth/login". The port should be set in the .env file.

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -110,43 +110,43 @@ exports.create = function(req, res) {
...
-        }
-        if (!duplicate) {
-          console.log('Validated donation');
-          user.donations.push(req.body);
-          user.premium = 1;
-          user.save();
-        }
-      });
+          if (!duplicate) {
+            console.log('Validated donation');
```
Comment (adebayo-maborukoje) [inline app/controllers/users.js]: remove console.log in your code.

File: config/routes.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
-var async = require('async');
+'use strict';
 
-module.exports = function(app, passport, auth) {
+let async = require('async');
```
Comment (adebayo-maborukoje) [inline config/routes.js]: ### NIT PICK
Remove unused variables

File: test/server/user/auth.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      if(err) {
+        throw err;
+      }
+      done();
+    });
+  });
+  describe('Login', () => {
+    it('should return an error on the wrong username or password login',
+      (done) => {
+        let user = {
```
Comment (adebayo-maborukoje) [inline test/server/user/auth.js]: ### NIT PICK
Change to `const`

File: test/server/user/auth.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          username: 'user1',
+          password: 'password'
+        };
+
+        chai.request(server)
+          .post('/api/auth/login')
+          .send(user)
+          .end((err, res) => {
+            res.body.should.be.a('object');
```
Comment (adebayo-maborukoje) [inline test/server/user/auth.js]: Make an assertion for the status code being returned

Comment (andela-emurphy) [inline test/server/user/auth.js]: the status code test has been  added

File: app/controllers/auth.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        message: 'Auhentication failed. invalid details'
+      });
+    }
+    const token = jwt.sign({
+      user: user.email}, config.secret ,{
+        expiresIn: moment().add(7, 'd').valueOf()
+      });
+    return res.status(200).json({
+      success: true,
+      message: 'login successfull',
```
Comment (andela-emurphy) [inline app/controllers/auth.js]: i corrected the spellings

File: app/controllers/users.js (hunk 1 above)
Comment (andela-emurphy) [inline app/controllers/users.js]: I removed the console.log

Comment (andela-emurphy) [inline app/controllers/users.js]: I removed the console.log

File: test/server/user/auth.js (hunk 3 above)
Comment (andela-emurphy) [inline test/server/user/auth.js]: i changed the let to const

**Pull request merged**